tests/parse_metadata.py

```diff
@@ -28,8 +30,13 @@
     "read_dependencies",
     "read_metadata",
     "read_stubtest_settings",
+    "OLDEST_SUPPORTED_PYTHON",
+    "PYTHON_VERSION",
 ]
 
+with open("pyproject.toml", "rb") as config:
+    OLDEST_SUPPORTED_PYTHON: Final[str] = tomli.load(config)["tool"]["typeshed"]["oldest_supported_python"]
```

Comment:
`OLDEST_SUPPORTED_PYTHON` is only used in the `read_metadata` function. Should I move it there?

Reply:
That feels like it would be inefficient (we'd have to reload it from the `pyproject.toml` file every time `read_metadata` was called with a different argument). But I think I'd prefer to do something like this -- not massively keen on files being opened and read from just as a side effect of importing a module:

```py
@cache
def _get_oldest_supported_python() -> str:
    with open("pyproject.toml", "rb") as config:
        val = tomli.load(config)["tool"]["typeshed"]["oldest_supported_python"]
    assert type(val) is str
    return val
```

And then in `read_metadata()` you can just do

```py
oldest_supported_python = _get_oldest_supported_python()
```

Reply:
We could possibly make `_get_oldest_supported_python()` public at a later date if we find it's useful from other files in our `tests/` directory, like @Avasam suggests